### PR TITLE
Vectorizer, take 2

### PR DESCRIPTION
**Pull request by resistor**

### CODE REVIEW

File: test/cpp/tensorexpr/test_llvm.cpp [hunk 1]
```diff
@@ -243,6 +243,7 @@ void testLLVMLoadStoreTest() {
   EXPECT_EQ(b_buffer[0], 42);
 }
 
+
```
Comment (bertmaher) [inline test/cpp/tensorexpr/test_llvm.cpp]: nit - extraneous whitespace

File: torch/csrc/jit/tensorexpr/ir_mutator.h [hunk 2]
```diff
@@ -90,6 +91,9 @@ AT_FORALL_SCALAR_TYPES_AND2(Bool, Half, IMM_MUTATE_DECLARE);
   virtual Stmt* mutate(const Allocate* v);
   virtual Stmt* mutate(const Free* v);
   virtual Stmt* mutate(const Cond* v);
+
+  protected:
+  const Expr* DefaultMutator(const BaseCallNode* v, std::vector<const Expr*>& params);
```
Comment (bertmaher) [inline torch/csrc/jit/tensorexpr/ir_mutator.h:96]: We're not consistent about naming currently but I think we should try to get back the fbcode convention of `lowerCamelCase` for methods.

File: torch/csrc/jit/tensorexpr/schedule.cpp [hunk 3]
```diff
@@ -392,6 +392,237 @@ ScheduleObject* ScheduleNode::CloneScheduleObject(ScheduleObject* object) {
...
+  Stmt* vectorize(const For* v) {
+    Stmt* body = v->body();
+    const Var* var = v->var();
+    const Expr* start = v->start();
+    const Expr* stop = v->stop();
+
+    const IntImm* start_imm = dynamic_cast<const IntImm*>(start);
+    const IntImm* stop_imm = dynamic_cast<const IntImm*>(stop);
+    if (!start_imm) {
+      throw "Can't vectorize due to non-constant loop start!";
```
Comment (bertmaher) [inline torch/csrc/jit/tensorexpr/schedule.cpp]: Throwing strings seems kind of sketch.  What's pytorch convention for exceptions?  Should we just use `std::runtime_error`?

**Review by bertmaher — APPROVED** [review]

File: torch/csrc/jit/tensorexpr/ir_mutator.h (hunk 2 above)
Comment (resistor) [reply torch/csrc/jit/tensorexpr/ir_mutator.h:96]: It's currently using this convention to match `BaseCallNode::DefaultMutator`.  I'm fine with changing the convention but I'd rather do that separately.

**Pull request merged**